### PR TITLE
Add TraceDataCollectorDirectoryPath parameter to VSTestTask

### PR DESCRIPTION
**Pull request by smadala**
## Description
Value for TraceDataCollectorDirectoryPath  will be passed from targets present in Microsoft.CodeCoverage package. [More details](https://microsoft.sharepoint.com/teams/DD_TestingTools/_layouts/OneNote.aspx?id=%2Fteams%2FDD_TestingTools%2FDocuments%2FUT%20CUIT%2Fdev15%2FVSTest%20%28core%29&wd=target%28Design.one%7C5F330F1B-7783-402D-9305-16093F37EA63%2FSpike%20for%20automatically%20discovery%20of%20TraceDataCollector%28TDC%5C%29%7C71E08665-1538-4C92-B78D-8619932C9931%2F%29)

## Related issue
#981

### CODE REVIEW

File: test/Microsoft.TestPlatform.Build.UnitTests/VsTestTaskTests.cs [hunk 1]
```diff
@@ -255,19 +198,50 @@ public void CreateArgumentShouldAddMultipleTestAdapterPaths()
...
+            this.vsTestTask.VSTestCollect = new string[] { "not code coverage" };
+
+            var allArguments = this.vsTestTask.CreateArgument().ToArray();
+
+            const string notExpectedArg = "--testAdapterPath:\"c:\\path\\to\\tracedata collector\"";
+            CollectionAssert.DoesNotContain(allArguments, notExpectedArg, $"Not expected argument: '''{notExpectedArg}''' present in [{string.Join(", ", allArguments)}]");
+        }
+
+        [TestMethod]
+        public void CreateArgumentShouldAddTraceCollectorDirectoryPathAsTestAdapterIfSettingsGiven()
```
Comment (mayankbansal018) [inline test/Microsoft.TestPlatform.Build.UnitTests/VsTestTaskTests.cs]: Nit: Check if we can add another test where VSTestTraceDataCollectorDirectoryPath is empty, so it is not set if if runsettings is present

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 2]
```diff
@@ -233,10 +242,25 @@ internal IEnumerable<string> CreateArgument()
...
+                    {
+                        isCollectCodeCoverageEnabled = true;
+                    }
                     allArgs.Add("--collect:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
                 }
             }
 
+            if (isCollectCodeCoverageEnabled || isRunSettingsEnabled)
+            {
+                // Pass TraceDataCollector path to vstest.console as TestAdapterPath if --collect "Code Coverage"
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: Here trade off is b/w always parsing runsettings(which we don't want to do), vs always adding the path of TraceDataCollector dir if it is present, & runsettings is used.

I agree with latter.

**Review by mayankbansal018 — APPROVED** [review]
:shipit:

File: test/Microsoft.TestPlatform.Build.UnitTests/VsTestTaskTests.cs (hunk 1 above)
Comment (smadala) [reply test/Microsoft.TestPlatform.Build.UnitTests/VsTestTaskTests.cs]: Done.

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs (hunk 2 above)
Comment (smadala) [reply src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: Done.
